docs/configuration_files.rst

```diff
@@ -0,0 +1,350 @@
+      mylist: [ 'abc', 789, 2.0e3 ] # A list
+      mydict: { 'fruit': 'orange', 'year': 2020 } # A dictionary
+
+      # SkyPy extra functionality
+      angle: 10 deg
+      distance: 300 kpc
+
+
+Functions
+^^^^^^^^^
+* `Call a function`: functions are defined as tuples where the first entry is the fully qualified function name tagged with and exclamation mark ``!`` and the second entry is either a list of positional arguments or a dictionary of keyword arguments.
+
+  For example, if you need to call the ``log10()`` and ``linspace()`` NumPy_ functions, then you define the following key-value pairs:
+
+  .. code:: yaml
+
+      log_of_2: !numpy.log10 [2]
+      myarray: !numpy.linspace [0, 2.5, 10]
+
+  You can also define parameters of functions with a dictionary of keyword arguments.
+  Imagine you want to compute the comoving distance for a range of redshifts and an `Astropy` Planck 2015 cosmology.
+  To run it with the `SkyPy` pipeline, call the function and define the parameters as an indented dictionary.
+
+  .. code:: yaml
+
+      comoving_distance: !astropy.cosmology.Planck15.comoving_distance
+        z: !numpy.linspace [ 0, 1.3, 10 ]
+
+  Similarly, you can specify the functions arguments as a dictionary:
+
+  .. code:: yaml
+
+      comoving_distance: !astropy.cosmology.Planck15.comoving_distance
+        z: !numpy.linspace {start: 0, stop: 1.3, num: 10}
+
+  `N.B.` To call a function with no arguments, you should pass an empty list of
+  ``args`` or an empty dictionary of ``kwargs``. For example:
+
+  .. code:: yaml
+
+      cosmo: !astropy.cosmology.default_cosmology.get []
+
+
+* `Reference a variable`: variables can be referenced by their full name tagged with a dollar sign ``$``.
+  In the previous example you could also define the variables at the top-level of the file and then reference them:
+
+  .. code:: yaml
+
+      redshift: !numpy.linspace [ 0, 1.3, 10 ]
+      comoving_distance: !astropy.cosmology.Planck15.comoving_distance
+        z: $redshift
+
+  Please, check below in the `cosmology`_ section how to use different cosmologies with the SkyPy pipeline.
+
+
+Tables
+^^^^^^
+
+* `Create a table`: a dictionary of table names, each resolving to a dictionary of column names for that table.
+
+  Let us create a table called ``telescope`` with a column to store the width of spectral lines that follow a normal distribution
+
+  .. code:: yaml
+
+      tables:
+        telescope:
+          spectral_lines: !scipy.stats.norm
+            loc: 550
+            scale: 1.6
+            size: 100
+
+* `Add a column`: you can add as many columns to a table as you need.
+    Imagine you want to add a column for the telescope collecting surface
+
+  .. code:: yaml
+
+      tables:
+        telescope:
+          spectral_lines: !scipy.stats.norm
+            loc: 550
+            scale: 1.6
+            size: 100
+          collecting_surface: !numpy.random.uniform
+            low: 6.9
+            high: 7.1
```

Comment:
```suggestion
            high: 7.1
            size: 100
```
This example doesn't work because the columns don't have the same length; is this what you intended?